src/main/java/org/jabref/gui/search/GlobalSearchResultDialog.java

```diff
@@ -0,0 +1,126 @@
+package org.jabref.gui.search;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.swing.undo.UndoManager;
+
+import javafx.collections.ObservableList;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.DialogPane;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.externalfiletype.ExternalFileTypes;
+import org.jabref.gui.groups.GroupViewMode;
+import org.jabref.gui.maintable.MainTableColumnModel;
+import org.jabref.gui.maintable.MainTableColumnModel.Type;
+import org.jabref.gui.maintable.MainTableDataModel;
+import org.jabref.gui.maintable.columns.FieldColumn;
+import org.jabref.gui.maintable.columns.SpecialFieldColumn;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.search.SearchQuery;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.UnknownField;
+import org.jabref.model.groups.GroupTreeNode;
+import org.jabref.model.search.matchers.MatcherSet;
+import org.jabref.model.search.matchers.MatcherSets;
+import org.jabref.preferences.PreferencesService;
+
+public class GlobalSearchResultDialog {
+
+    public static String LIBRARY_NAME_FIELD = "Library_Name";
+
+    private final ExternalFileTypes externalFileTypes;
+    private final PreferencesService preferencesService;
+    private final StateManager stateManager;
+    private final BibDatabaseContext context;
+    private final DialogService dialogService;
+    private final FieldColumn libColumn;
+    private final UndoManager undoManager;
+    private final GroupViewMode groupViewMode;
+
+    public GlobalSearchResultDialog(PreferencesService preferencesService, StateManager stateManager, ExternalFileTypes externalFileTypes, UndoManager undoManager, DialogService dialogService) {
+        this.context = new BibDatabaseContext();
+        this.preferencesService = preferencesService;
+        this.stateManager = stateManager;
+        this.externalFileTypes = externalFileTypes;
+        this.undoManager = undoManager;
+        this.dialogService = dialogService;
+
+        this.groupViewMode = preferencesService.getGroupViewMode();
+        this.libColumn = new FieldColumn(new MainTableColumnModel(Type.NORMALFIELD, LIBRARY_NAME_FIELD));
+    }
+
+    public void doGlobalSearch() {
+        if (stateManager.isGlobalSearchActive()) {
+
+            BibDatabaseContext resultDbContext = new BibDatabaseContext();
+
+            for (BibDatabaseContext dbContext : this.stateManager.getOpenDatabases()) {
+
+                List<BibEntry> result = dbContext.getEntries().stream().filter(entry -> isMatched(stateManager.activeGroupProperty(), stateManager.activeSearchQueryProperty().get(), entry))
```

Comment:
better pass the search query as a parameter? Feels cleaner in my opinion, and you don't rely on the fact that doGlobalSearch needs to be called after the state property is updated.

Reply:
Does it really make sense to filter the entries by the active group for the global search? I would expect that this would result in a very counter-intuitive behavior.

Reply:
Good point